### PR TITLE
[10.0][IMP] web_timeline change the option to initialize the timeline window on display.

### PR DESCRIPTION
**Pull request by Creamaster**
Change the attribute '**default_window**' to '**mode**' like the Odoo calendar view (to be iso functional)

### CODE REVIEW

**Comment by Creamaster** [comment]
This PR contains a commit I initially intended to include in a previous PR ([#735](https://github.com/OCA/web/pull/735)). But was forgotten in my local branch, while I was working on a rework of the group_by functionality, to address the issue [OCA/project/#316.](github.com/OCA/project/issues/316)

This PR is a very small improvement/correction and can be rejected as the time needed to validate a PR could be greater than it's gain.

With this change, now a display mode 'day' show the current day, not the next 24h

**Comment by pedrobaeza** [comment]
Please change module version number, and of course this is very welcome!

**Comment by Creamaster** [comment]
Module version number updated 10.0.1.1.0 -->  10.0.1.2.0

**Review by pedrobaeza — APPROVED** [review]
@Yajo please review

File: web_timeline/static/src/js/web_timeline.js [hunk 1]
```diff
@@ -111,7 +104,7 @@ odoo.define('web_timeline.TimelineView', function (require) {
             this.date_delay = attrs.date_delay;
             this.no_period = this.date_start == this.date_stop;
             this.zoomKey = attrs.zoomKey || '';
-            this.default_window = attrs.default_window || 'fit';
+            this.mode = attrs.mode || 'fit';
```
Comment (yajo) [inline web_timeline/static/src/js/web_timeline.js]: The rest of the patch is OK, but I see no reason for breaking backwards compatibility by changing the option name. Can't we leave the `default_window` name as it was before?

File: web_timeline/static/src/js/web_timeline.js [hunk 2]
```diff
@@ -172,25 +165,27 @@ odoo.define('web_timeline.TimelineView', function (require) {
                 onRemove: self.on_remove,
                 zoomKey: this.zoomKey
             };
-            if (this.default_window) {
-                var start = new moment();
-                var end;
-                switch (this.default_window) {
+            if (this.mode) {
+                var start, end = false;
```
Comment (yajo) [inline web_timeline/static/src/js/web_timeline.js]: Any reason why `start` defaults to `undefined` and `end` defaults to `false`?

**Review by yajo — CHANGES_REQUESTED** [review]

File: web_timeline/static/src/js/web_timeline.js (hunk 1 above)
Comment (Creamaster) [reply web_timeline/static/src/js/web_timeline.js]: I added the attribute 'default_window' just before this PR. this PR intend to change the 'default_window' behavior, to a more usable one.
The new parameter is the same as the default Odoo one (same functionality), and to facilitate the use of the timeline view, it seems best to use the same attribute name for the same behavior in similar view ([see Odoo calendar view default view mode](https://github.com/odoo/odoo/blob/10.0/addons/web_calendar/static/src/js/web_calendar.js#L127))

I will change the attribute to be backwards compatible, using both syntax, like : 
```javascript
this.mode = attrs.mode || attrs.default_window || 'fit';
```
Is it okay ?

File: web_timeline/static/src/js/web_timeline.js (hunk 2 above)
Comment (Creamaster) [reply web_timeline/static/src/js/web_timeline.js]: No particular reasons, I often declare "null" variables like that for readability.
I will change it with :
```javascript
var start = false, end = false;
```
To be more explicit

File: web_timeline/static/src/js/web_timeline.js (hunk 1 above)
Comment (yajo) [reply web_timeline/static/src/js/web_timeline.js]: In such case, then please break backwards compatibility 😊

File: web_timeline/static/src/js/web_timeline.js (hunk 2 above)
Comment (yajo) [reply web_timeline/static/src/js/web_timeline.js]: `undefined` is a falsey value, so if it doesn't really matter, just do `var start, end;`

**Review by simahawk — APPROVED** [review]

**Comment by simahawk** [comment]
@Yajo sounds like your comments were addressed: can you check again?

**Review by yajo — APPROVED** [review]

**Pull request merged**